test/relations.test.js

```diff
@@ -2569,7 +2569,7 @@ describe('relations', function () {
       Person.create({ name: 'Wilma', addresses: addresses }, function(err, p) {
         err.name.should.equal('ValidationError');
         var expected = 'The `Person` instance is not valid. ';
-        expected += 'Details: `addresses` contains invalid item: `work` (`street` can\'t be blank).';
+        expected += 'Details: `addresses: [ { id: \u001b[32m\'home\'\u001b[39m, street: \u001b[32m\'Home Street\'\u001b[39m },\n  { id: \u001b[32m\'work\'\u001b[39m, street: \u001b[32m\'\'\u001b[39m } ]` contains invalid item: `work` (`street` can\'t be blank).';
```

Comment:
Ok.  Why are the color codes getting inserted in the first place?  I'm using 'inspect' from node core API.  I had never noticed it to print colors before.  Is one of the Loopback dependencies monkey patching it to add colors?

I decided to use 'inspect' because it limits how deep it will parse the tree.  This seemed like desirable behavior which JSON.stringify does not offer.

> On Jan 8, 2015, at 1:03 AM, Miroslav Bajtoš <notifications@github.com> wrote:
> 
> In test/relations.test.js:
> 
> > @@ -2569,7 +2569,7 @@ describe('relations', function () {
> >        Person.create({ name: 'Wilma', addresses: addresses }, function(err, p) {
> >          err.name.should.equal('ValidationError');
> >          var expected = 'The `Person` instance is not valid. ';
> > -        expected += 'Details: `addresses` contains invalid item: `work` (`street` can\'t be blank).';
> > +        expected += 'Details: `addresses: [ { id: \u001b[32m\'home\'\u001b[39m, street: \u001b[32m\'Home Street\'\u001b[39m },\n  { id: \u001b[32m\'work\'\u001b[39m, street: \u001b[32m\'\'\u001b[39m } ]` contains invalid item: `work` (`street` can\'t be blank).';
> Notice how an invalid item in an array is handled: the message contains id of the item, and then information about the error in this item only.
> 
> This patch should build on top of that and print only the invalid item in the message. Example:
> 
> On the second thought, that would be a much more involved change, let's leave it for another day.
> 
> However, you still need to remove colour codes.
> 
> —
> Reply to this email directly or view it on GitHub.
>